### ISSUE TITLE
[Bug] MPS Training Fails with NaN Loss on M4/M-Series (Root Cause Identified)

### ISSUE THREAD

**Issue by jucrowle** [instructlab/instructlab#3533]
### Summary
InstructLab training consistently fails with `NaN` loss on Step 1 when using the MPS (Apple Silicon) device on newer chips (M4 Pro).

I have identified the root cause: PyTorch's default `use_reentrant=True` in gradient checkpointing interacts poorly with the MPS backend on these devices.

### Environment
- **Hardware:** Mac Mini M4 Pro (64GB RAM)
- **OS:** macOS 15.2 (Beta)
- **InstructLab Version:** 0.26.1
- **PyTorch Version:** 2.7.1 (Nightly/Preview)

### Root Cause Analysis
The default implementation of `gradient_checkpointing_enable()` uses `use_reentrant=True`.
On MPS, this causes non-contiguous tensor handling issues during the backward pass, leading immediately to `NaN` gradients.

Switching to `use_reentrant=False` (the newer PyTorch implementation) completely resolves the issue.

### Reproduction Script
<details>
<summary>Click to see reproduction script</summary>

```python
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer
import os

def test_checkpointing(use_reentrant_val):
    device = torch.device("mps")
    # Use any available local model path
    model_path = os.path.expanduser("~/.cache/instructlab/models/instructlab/granite-7b-lab")
    
    model = AutoModelForCausalLM.from_pretrained(model_path, torch_dtype=torch.float32)
    model.to(device)
    model.gradient_checkpointing_enable(gradient_checkpointing_kwargs={"use_reentrant": use_reentrant_val})
    
    # Simple forward pass test...
    # (The script demonstrates NaN with True, and success with False)
</details>

Proposed Fix

In instructlab/model/full_train.py, the gradient checkpointing call should be patched for MPS devices:

Python
if dev.type == "mps":
    # PATCH: Fix NaN on MPS by disabling reentrant checkpointing
    model.gradient_checkpointing_enable(
        gradient_checkpointing_kwargs={"use_reentrant": False}
    )
else:
    model.gradient_checkpointing_enable()
Verification

After applying this patch locally, I was able to train successfully on M4 Pro:

Before: NaN loss on Step 1.

After: Real loss values; training completes successfully.

Performance: ~6x faster than CPU fallback.
